src/inmanta/agent/executor.py

```diff
@@ -186,10 +195,10 @@ def to_env_blueprint(self) -> EnvBlueprint:
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, ExecutorBlueprint):
             return False
-        return (self.pip_config, set(self.requirements), sorted(self.sources)) == (
+        return (self.pip_config, set(self.requirements), sorted(list(set(self.sources)))) == (
```

Comment:
Shouldn't we check simple equality now that we sort these at construction? i.e. `@dataclass(frozen=True, eq=True)`? May have to use `__setattr__` if you go for frozen though: https://docs.python.org/3/library/dataclasses.html#frozen-instances.